server/bootstrap/passport/strategies/bnet.js

```diff
@@ -8,7 +8,7 @@ module.exports = function(authBnetConfig){
     function(accessToken, refreshToken, profile, done) {
       if (accessToken !== null && profile.id > 0) {
         User
-          .createOrUpdate(profile.id, accessToken, "bnet")
+          .createOrUpdate(profile.id, profile.battletag, accessToken, "bnet")
```

Comment:
We should move this functionality to the `BattlenetUser` module, since it's very specific to registration via Battle.net. Ideally, the `User` module should be an access point which works regardless of how the user was created.